apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -3,28 +3,43 @@ package v1alpha1
 import (
 	"fmt"
 
-	"k8s.io/apimachinery/pkg/util/errors"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 // log is for logging in this package.
 var log = logf.Log.WithName("baremetalhost-validation")
 
-func (host *BareMetalHost) Validate(old *BareMetalHost) error {
-	log.Info("validate", "name", host.Name)
+// validateCreate validates BareMetalHost resource for creation
+func (host *BareMetalHost) validateCreate() []error {
+	log.Info("validate create", "name", host.Name)
 	var errs []error
 
 	if err := validateRAID(host.Spec.RAID); err != nil {
 		errs = append(errs, err)
 	}
 
-	if old != nil {
-		if updateErrs := validateUpdate(old, host); updateErrs != nil {
-			errs = append(errs, updateErrs...)
-		}
+	return errs
+}
+
+// validateUpdate validates BareMetalHost resource for update
+// but also covers the validations of creation
```

Comment:
Maybe we should call them validateHost() and validateChanges() or something like that?